homeassistant/components/ssdp/server.py

```diff
@@ -99,16 +96,16 @@ async def _async_find_next_available_port(
     with ExitStack() as stack:
         test_socket = stack.enter_context(socket.socket(family, socket.SOCK_STREAM))
         test_socket.setblocking(False)
-        test_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
 
-        for port in range(UPNP_SERVER_MIN_PORT, UPNP_SERVER_MAX_PORT):
-            addr = (source[0], port, *source[2:])
+        for i in range(100):
```

Comment:
We don't need to retry here. If there are no ports free or something seriously wrong we shouldn't keep trying as its unlikely to ever be successful.